### ISSUE TITLE
char_to_token() is not available when using Python based tokenizers

### ISSUE THREAD

**Issue by OanaIgnat** [ashkamath/mdetr#86]
Hi, I am using your requirements file, same version of transformers, and for the pretraining part I get this error:
char_to_token() is not available when using Python based tokenizers

**Comment by Ngheissari** [comment]
I have the same issue. Any progress ?

**Comment by shikunyu8** [comment]
I get this error as well.

**Comment by alcinos** [comment]
This error most likely indicates divergence in your model. You can check by adding asserts like:
```
assert not boxes1.isnan().any().item(), "nan in boxes1"
```
in the `generalized_box_iou` function which is currently failing.

If this is established, there could be several reasons that is happening, in order of likelihood:

1. Total batch-size is not big enough. We trained with total batch size = 64 (achieved with batch 2 per gpus and 4 nodes of 8 gpus). Training is possibly going to be stable at bs 32, and _maybe_ at 16 but lower than that is unlikely to work
2. Mixed precision is tricky to get working. If you tried some sort of AMP autocast or FP16 training that could be the reason for the divergence
3. Corruption in your training data that somehow triggers gradient explosion.
4. In some rare cases, if the weight decay is too high it can create nans as well, although it is unlikely in that case because it happens early in training.


Hope this helps

**Comment by shikunyu8** [comment]
I got the error when running inference with the released checkpoint of MDETR-EB3 on RefCOCO.

**Comment by conan1024hao** [comment]
same problem

**Comment by Franklin905** [comment]
I got the same problem too when using the pre-trained MDETR-EB5 from torch.hub during inference. Does anyone know how to fix this problem?

**Comment by lclszsdnr** [comment]
> I got the error when running inference with the released checkpoint of MDETR-EB3 on RefCOCO.

Hi,Do you have checked the img feature from EB3,in my case the feature is very large ,sometimes is NAN

**Comment by BatmanofZuhandArrgh** [comment]
> This error most likely indicates divergence in your model. You can check by adding asserts like:
> 
> ```
> assert not boxes1.isnan().any().item(), "nan in boxes1"
> ```
> 
> in the `generalized_box_iou` function which is currently failing.
> 
> If this is established, there could be several reasons that is happening, in order of likelihood:
> 
> 1. Total batch-size is not big enough. We trained with total batch size = 64 (achieved with batch 2 per gpus and 4 nodes of 8 gpus). Training is possibly going to be stable at bs 32, and _maybe_ at 16 but lower than that is unlikely to work
> 2. Mixed precision is tricky to get working. If you tried some sort of AMP autocast or FP16 training that could be the reason for the divergence
> 3. Corruption in your training data that somehow triggers gradient explosion.
> 4. In some rare cases, if the weight decay is too high it can create nans as well, although it is unlikely in that case because it happens early in training.
> 
> Hope this helps

Hi I'm trying FP16 and running evaluation on Flickr30k val set. Its output is also giving "nan", only for few samples though. What is this divergence issue you're speaking of? @alcinos
